### DISCUSSION TITLE
Can I control the switching of outputs via scripting?

### DISCUSSION THREAD

**Discussion by birdjumper** [kyleneideck/BackgroundMusic#749] [Q&A] [answered]
Hi folks

I spend a lot of my time switching between inputs and it would be convenient if I was able to do that via some kind of scripting.

Is there anything available to do this or is there a way that other people are doing it? 

Thanks

birdjumper

Mac M1 RAM 16GB running macOS Sonoma v14.5, Background Music Version 0.4.0-SNAPSHOT-b38f6dd

**Comment by kyleneideck** [comment, marked as answer]
Does this help? https://github.com/kyleneideck/BackgroundMusic/issues/278#issuecomment-626118939

**Reply by birdjumper** [reply]
Hi Kyle

Thanks, this works very well for me in Applescript:

tell application "Background Music"
	set selected output device to (output device "External Headphones")
end tell

Much appreciated.
